Doc/whatsnew/3.13.rst

```diff
@@ -478,6 +486,122 @@ Optimizations
   (Contributed by Jakub Kulik in :gh:`113117`.)
 
 
+.. _whatsnew313-jit-compiler:
+
+Experimental JIT Compiler
+=========================
+
+:Editor: Guido van Rossum, Ken Jin
+
+When CPython is configured using the ``--enable-experimental-jit`` option,
+a just-in-time compiler is added which can speed up some Python programs.
+The internal architecture is roughly as follows.
+
+Intermediate Representation
+---------------------------
+
+We start with specialized *Tier 1 bytecode*.
+See :ref:`What's new in 3.11 <whatsnew311-pep659>` for details.
+
+When the Tier 1 bytecode gets hot enough, the interpreter creates
+straight-line sequences of bytecode known as "traces", and translates that
+to a new, purely internal *Tier 2 IR*, a.k.a. micro-ops ("uops").
+These straight-line sequences can cross function call boundaries,
+allowing more effective optimizations, listed in the next section.
+
+The Tier 2 IR uses the same stack-based VM as Tier 1, but the
+instruction format is better suited to translation to machine code.
+
+(Tier 2 IR contributed by Mark Shannon and Guido van Rossum.)
+
+Optimizations
+-------------
+
+We have several optimization and analysis passes for Tier 2 IR, which
+are applied before Tier 2 IR is interpreted or translated to machine code.
+These optimizations take unoptimized Tier 2 IR and produce optimized Tier 2
+IR:
+
+* Type propagation -- through forward data-flow analysis, we infer
+  and deduce information about types. This allows us to eliminate
+  much of the overhead associated with dynamic typing in the future.
+
+* Constant propagation -- through forward data-flow analysis, we can reduce
+  expressions like ::
+
+    a = 1
+    b = 2
+    c = a + b
+
+ to ::
+
+    a = 1
+    b = 2
+    c = 3
+
+* Guard elimination -- through a combination of constant and type information,
```

Comment:
I wonder if the guard concept could be expanded a bit more. The example of "type checking" makes things fairly clear what it means, but wonder if it could be expanded.